scenarios.md

```diff
@@ -0,0 +1,190 @@
+
+**Implications of this requirement:**
+
+- In addition to the above scenario where an artifact is promoted within the same registry, this scenario requires the signature to be avaialble in the new registry. Whether the signature is moved, or referenced from it's original location is a subject for design discussion. In this scenario, ACME Rockets could validate the signature against a wabbit-network signature server.
+
+### Scenario #6: Validate Artifact Signatures Within Restricted Networks
+
+ACME Rockets runs secure production environments, limiting all external network traffic. To assure the wabbit-networks network monitor software has valid signatures, they will need to trust a resource within their network to proxy the signature validations.
+
+1. ACME Rockets acquires network monitoring software, copying it to their firewall protected production environment.
+1. As part of the artifact copy, they will copy/proxy the signature validation to trusted resources within their network protected environment.
+
+**Implications of this requirement:**
+
+- In this scenario, the wabbit-netowrks signature must be validated within the ACME Rockets network. How this is done is open for design. However, the requirement states the signature must be validated without external access. When the artifact is copied to the private/network restricted registry, the signature may need to be copied, and is assumed to be trusted if available in the trusted server within the private network. 
+
+### Scenario #7 Automated Deploy - Success
+
+A CD solution initiates a deployment to a development environment. The development environment has been previously configured to only accept signed artifacts.
+
+1. A CD solution is triggered to initiate a deployment
+1. The CD solution uses a deployment template to request an orchestrator to run specific images.
+1. The orchestrator receives the request, validates the artifact is signed and the signature is valid *before* it can initiate the deployment.
+1. The signature is considered valid, and the deployment is successful.
+
+**Implications of this requirement:**
+
+- The aspect of being a signed artifact is not germane to the CD solution. The CD solution simply requests a deployment. The enforcement of signature validation is a policy of the execution environment.
+
+### Scenario #8 Automated Deploy - Failure
+
+Same as #7, with the exception that the signature is deemed invalid.
+1. A CD solution is triggered to initiate a deployment.
+2. The CD solution uses a deployment template to request an orchestrator to run a specific image. In this case, the CD solution mistakenly attempted to deploy a dev signed artifact to a production environment.
+3. The production orchestrator receives the request, validates the artifact is signed and recognizes the signature is **NOT** one of the production accepted signatures. As the signature is deemed invalid, the deployment fails.
```

Comment:
I'd replace "one of the production accepted signatures" with "produced by a trusted key".